src/metabase/query_processor/streaming/xlsx.clj

```diff
@@ -589,106 +587,100 @@
         typed-cell-styles           (compute-typed-cell-styles wb data-format)
         data-sheet                  (spreadsheet/select-sheet "data" wb)
         pivot-sheet                 (spreadsheet/select-sheet "pivot" wb)
-        area-ref                    (cell-range->area-ref (cell-range rows))
-        _                           (doseq [row rows]
-                                      (add-row! data-sheet row ordered-cols col-settings cell-styles typed-cell-styles))
+        col-names                   (common/column-titles ordered-cols col-settings format-rows?)
+        _                           (add-row! data-sheet col-names ordered-cols col-settings cell-styles typed-cell-styles)
+        area-ref                    (AreaReference/getWholeColumn SpreadsheetVersion/EXCEL2007
+                                                                  "A"
+                                                                  (CellReference/convertNumToColString (dec (count ordered-cols))))
         ^XSSFPivotTable pivot-table (.createPivotTable ^XSSFSheet pivot-sheet
                                                        ^AreaReference area-ref
-                                                       (CellReference. "A1")
+                                                       (CellReference. 0 0)
                                                        ^XSSFSheet data-sheet)]
     (doseq [idx pivot-rows]
       (.addRowLabel pivot-table idx))
     (doseq [idx pivot-cols]
       (.addColLabel pivot-table idx))
     (doseq [idx pivot-measures]
-      (.addColumnLabel pivot-table (get aggregation-functions idx DataConsolidateFunction/COUNT) idx))
-    wb))
-
-;; As a first step towards hollistically solving this issue: https://github.com/metabase/metabase/issues/44556
-;; (which is basically that very large pivot tables can crash the export process),
-;; The post processing is disabled completely.
-;; This should remain `false` until it's fixed
-;; TODO: rework this post-processing once there's a clear way in app to enable/disable it, or to select alternate download options
-(def ^:dynamic *pivot-export-post-processing-enabled*
-  "Flag to enable/disable export post-processing of pivot tables.
-  Disabled by default and should remain disabled until Issue #44556 is resolved and a clear plan is made."
-  false)
+      (.addColumnLabel pivot-table DataConsolidateFunction/SUM #_(get aggregation-functions idx DataConsolidateFunction/SUM) idx))
+    (let [swb   (-> (SXSSFWorkbook. ^XSSFWorkbook wb)
+                    (doto (.setCompressTempFiles true)))
+          sheet (spreadsheet/select-sheet "data" swb)]
+      (doseq [i (range (count ordered-cols))]
+        (.trackColumnForAutoSizing ^SXSSFSheet sheet i))
+      (setup-header-row! sheet (count ordered-cols))
+      {:workbook swb
+       :sheet    sheet})))
+
+(defn- init-workbook
```

Comment:
Refactor the reguarl workbook setup stuff into its own function, both `init-workbook` and `init-pivot` output a map with the workbook and the sheet to add rows to.